src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -703,8 +703,20 @@ void AsynchronousMetrics::update(TimePoint update_time)
             Int64 free_memory_in_allocator_arenas = 0;
 
 #if USE_JEMALLOC
-            /// This is a memory which is kept by allocator.
-            /// Will subsract it from RSS to decrease memory drift.
+            /// According to jemalloc man, pdirty is:
+            ///
+            ///     Number of pages within unused extents that are potentially
+            ///     dirty, and for which madvise() or similar has not been called.
+            ///
+            /// So they will be subtracted from RSS to make accounting more
+            /// accurate, since those pages are not really RSS but a memory
+            /// that can be used at anytime via jemalloc.
+            ///
+            /// But as usual, things are not that simple as they should, since
+            /// ClickHouse allocates memory directly via mmap() too, and such
+            /// allocations cannot use this memory, so if you will have a
+            /// situation when you have lots of pdirty pages, some mmap() will
+            /// fail or OOM will kill the process.
```

Comment:
```suggestion
```

This is not strictly correct, since usually mmap() allocations will go via MemoryTracker